draft-hendrickson-pp-attesterissuer.md

```diff
@@ -124,32 +124,37 @@ same entity, as defined in section 4.3 of {{!RFC9576}}.
 ## Attester-to-Issuer Request {#request}
 
 The Attester and Issuer MUST use a mutually authenticated and secure HTTPS
-connection. They MAY use Mutual TLS for mutual aumutual authentication. The
-"IssuerTokenRequest" defined below is written in TLS Presentation Layer (Second
+connection. They MAY use Mutual TLS for mutual authentication. The
+"IssuerTokenRequest" defined below is written in TLS Presentation Layer (Section
 4 of {{!RFC8446}}), although the Attester and Issuer may instead choose to use
 another equivelent data interchange format such as JSON ({{!RFC8259}}).
 
-   struct {
-     uint16_t token_type;
-     uint8_t token_key_id;
-     uint8_t blinded_msg[Ne];
-   } IssuerTokenRequest;
+~~~tls
+struct {
+   uint16_t token_type;
+   opaque token<0..2^16-1>;
+} IssuerTokenRequest;
+~~~
 
-   The structure fields are defined as follows:
+The structure fields are defined as follows:
 
-   *  "token_type" is a 2-octet integer, which matches the type in the
-      challenge.
+- "token_type" is a 2-octet integer. TokenRequest MUST be prefixed with a uint16
+  "token_type" indicating the token type. The rest of the structure follows
+  based on that type, within the inner opaque token_request attribute. The above
+  definition corresponds to TokenRequest from {{RFC9578}}. For TokenRequest not
+  defined in {{RFC9578}}, they MAY be used as long as they are prefixed with a
+  2-octet token_type.
 
-   *  "token_key_id" which matches the type in the challenge.
+- "token" represents any required fields for the corresponding "token_type".
```

Comment:
This could be unclear. You don't want to repeat the token_type, and opaque will require a length prefix

Draft batched tokens uses select to handle this
https://github.com/ietf-wg-privacypass/ietf-draft-privacypass-batched-tokens/blob/main/draft-ietf-privacypass-batched-tokens.md?plain=1#L379

Reply:
Updated